### DISCUSSION TITLE
Please add Baidu Maps!

### DISCUSSION THREAD

**Discussion by belugaming** [teslamate-org/teslamate#3912] [Ideas]
Please add Baidu Maps API. Chinese users cannot use the default map source. I found Baidu Maps API. Please add it in the next version. We really need it. 
https://github.com/badafans/BaiduMap-Tiles-Download

**Comment by Amourgo** [comment]
As a Tesla owner in Chinese mainland, I urgently need to support the source data support of Baidu Maps or AutoNavi Maps.

**Reply by hipudding** [reply]
这个可以做，我这里有一个高德地图的[demo](https://github.com/hipudding/teslamate_fix_addrs/issues/7)，看是否能满足需求？

**Comment by gzjaychen** [comment]
Is it possible to use the Off-line OpenstreetMap data?
https://download.geofabrik.de/

**Comment by belugaming** [comment]
how to use

gzjaychen ***@***.***>于2024年6月12日 周三12:39写道：

**Reply by gzjaychen** [reply]
I think this will help you.
https://www.cnblogs.com/lin-bunny/p/15534670.html

**Comment by belugaming** [comment]
Yes, very helpful.

gzjaychen ***@***.***>于2024年6月12日 周三12:47写道：

**Reply by gzjaychen** [reply]
It would be great if teslamate provided a configurable openstreetmap url option.

**Comment by hipudding** [comment]
这个可以做，我这里有一个高德地图的[demo](https://github.com/hipudding/teslamate_fix_addrs/issues/7)，看是否能满足需求？
